### PR TITLE
Add first prototype of functionality to parse numbers in parantheses

### PR DESCRIPTION
**Pull request by onthebridgetonowhere**
### Related Issue #1737
Some initial thoughts on accepting numbers given as arguments in parentheses. 

Would love to hear some feedback if I am in the right direction or this is completely off. Currently, it does not support the error message improvement as per the suggestion from https://github.com/nushell/nushell/issues/1737#issuecomment-802292857. 
Needs proper testing, which I will add once I figure out if I am onto the right direction. 

### Questions
For the short style flags, what do we do with the number? Is that a positional argument? 

### Tried it out on calendar

`cal -y (-123) --> runs like cal -y`, so it ignores the (-123), but it gets parsed correctly as a `BigInt` with value `123` and minus sign.
`cal --full-year (2020) --> equivalent to cal --full-year 2020`
`cal --full-year (-2020) --> error, as cal does not accept negative years yet`

### Things to do:
- [x] allow whitespaces within (...) brackets
- [ ] do we need to support ranges?
- [x] fix the errors' styling, a bit confused there
- [ ] add tests

### CODE REVIEW

File: crates/nu-parser/src/parse.rs [hunk 1]
```diff
@@ -742,6 +742,34 @@ fn parse_table(
...
 }
 
+/// Parse the given argument in brackets () as a number
+fn parse_number_in_brackets(
+    expected_type: SyntaxShape,
+    scope: &dyn ParserScope,
+    lite_arg: &Spanned<String>,
+    ) -> (SpannedExpression, Option<ParseError>) {
+
+        if lite_arg.item.starts_with('(') && lite_arg.item.len() > 1 {
```
[the remark this answers is not in the document]
Comment (LhKipp) [reply crates/nu-parser/src/parse.rs:752]:
```rust
match (chars.next(), chars.next_back() {
}
```

[the remark this answers is not in the document]
Comment (onthebridgetonowhere) [reply crates/nu-parser/src/parse.rs:752]: Great, I will change to that. Thanks!

File: crates/nu-parser/src/parse.rs [hunk 2]
```diff
@@ -742,6 +742,34 @@ fn parse_table(
     )
 }
 
+/// Parse the given argument in brackets () as a number
+fn parse_number_in_brackets(
```
[the remark this answers is not in the document]
Comment (onthebridgetonowhere) [reply crates/nu-parser/src/parse.rs:746]: Agreed!

File: crates/nu-errors/src/lib.rs [hunk 3]
```diff
@@ -135,6 +135,8 @@ pub enum ArgumentError {
     MissingValueForName(String),
     /// An argument was found, but the command does not recognize it
     UnexpectedArgument(Spanned<String>),
+    /// An number argument has a missing closing bracket
+    MissingClosingBracket(String),
```
[the remark this answers is not in the document]
Comment (onthebridgetonowhere) [reply crates/nu-errors/src/lib.rs:139]: Definitely, thank you for the suggestion!

[the remark this answers is not in the document]
Comment (onthebridgetonowhere) [reply crates/nu-errors/src/lib.rs]: Removed these ones for now.

[the remark this answers is not in the document]
Comment (onthebridgetonowhere) [reply crates/nu-errors/src/lib.rs]: Removed these errors for now

**Comment by jonathandturner** [comment]
@onthebridgetonowhere - I was waiting for all four checkboxes, but looking now, do you want to have this PR reviewed as-is?

**Comment by onthebridgetonowhere** [comment]
@jonathandturner I guess I have not thought about ranges. Is this something we'd need at this point?
I should add tests, though I haven't had time to look into this in the past week or so. Should make more time for it soon. 

Would you prefer that I submit the tests before reviewing the PR? At the moment, all existing tests pass :-)

**Comment by jonathandturner** [comment]
Missed that you added the comment. Thanks for the PR!  cc @kubouch who is looking into some of the parser near here.

**Comment by kubouch** [comment]
@onthebridgetonowhere Thanks for your effort! I was thinking I could try implementing the parentheses support for ranges since I'm already refactoring the ranges in some commands. Unless you really want to do it yourself, of course.

**Comment by kubouch** [comment]
Also, the parentheses for ranges are not totally necessary since the following now works: `echo (-4)..-2`.

**Comment by onthebridgetonowhere** [comment]
@kubouch If you're looking into that area, then feel free to give it a try. I am not working on it at the moment. 
@jonathandturner my pleasure, thanks for merging it!

**Pull request merged**